lib/connectors_async/job_watcher.rb

```diff
@@ -0,0 +1,68 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+require 'concurrent'
+
+require 'connectors_async/job'
+require 'connectors_shared/logger'
+require 'connectors_shared/exception_tracking'
+
+module ConnectorsAsync
+  class JobWatcher
+    IDLE_TIME = 30 # seconds
+
+    class AlreadyWatchingError < StandardError; end
+
+    def initialize(job_store:)
+      @job_store = job_store
+
+      @is_watching = Concurrent::AtomicBoolean.new(false)
+    end
+
+    def watch
+      raise AlreadyWatchingError.new('Already watching!') unless @is_watching.make_true
+      ConnectorsShared::Logger.info('Watching after jobs {•̃_•̃}')
+
+      Thread.new do
+        loop do
+          run!
+        rescue StandardError => e
+          ConnectorsShared::ExceptionTracking.log_exception(e)
```

Comment:
so we catch the error and log it. and the thread exits. nothing else to do after that?

Reply:
I tried to cover it with the following test:
https://github.com/elastic/connectors/blob/57b24179042dba04a46870a585eeee28b4de8fef/spec/connectors_async/job_watcher_spec.rb#L54

In this example first call of `run!` raises a StandardError, and consecutive runs don't raise an error and do nothing. I've also set up `loop` to yield exactly 3 times, while in reality it yields infinitely, and the test checks that it's actually called 3 times.

So tldr - I log an error but the thread keeps running infinitely until the process is killed